File: lib/CreateRecursivePair/create_recursive_pair.ex
defmodule CreateRecursivePair do
     @moduledoc ~S"""
     #Create Recursive Pair
     Giving an array, create a clone for each value

    ## Examples

        iex> CreateRecursivePair.from([1,2])
        [1,1,2,2]
    """

    defp is_odd(arr, value) do
        Enum.count(Enum.filter(arr, fn(filtered_value) -> filtered_value == value end)) < 2
    end

    defp create_pairs(arr, value) do
        is_odd(arr,value) && arr ++ [value]
    end

    def from(range_list) do
        Enum.reduce(range_list, range_list, fn(curr, prev) -> create_pairs(prev, curr) end) |> Enum.sort()
    end
end

File: lib/CreateRecursivePair/other_languages/create_recursive_pair.js
var isOdd = (arr, value) => arr.filter(item => item === value).length < 2;
var createPair = (arr, value) => isOdd(arr, value) && arr.concat([value]);
var arrNumbers = [0,1,2,3,4];

arrNumbers.reduce((prev, curr, idx, arr) => createPair(prev,curr), arrNumbers).sort()